WordPress/gutenberg
# Show "none" as an alignment option and use contextual text to clarify settings

## Description
Resolves: https://github.com/WordPress/gutenberg/issues/34597
Fixes: https://github.com/WordPress/gutenberg/issues/30782


This PR is for now a rough POC to update the design of the alignment dropdown to include "none" as an option (it's the same as unsetting). The center icon has been updated, and the previous icon for centering has been repurposed for indicating "None". There's help text for "None" and "Wide", to show the values for the content and wide area layout values.

<div align="center"><img width="444" alt="Group 259" src="https://user-images.githubusercontent.com/1204802/132249337-49089891-a9d2-44c8-bd1c-db923a8fb031.png"></div>

For themes that do not support wide or fullwide alignments we render some contextual text to clarify the missing buttons:

<div align="center"><img width="444" alt="Group 260" src="https://user-images.githubusercontent.com/1204802/132249552-d45aafa8-dc32-4b80-8e6f-985cbb98c7bf.png"></div>

-- the above description is from the issue

## Testing instructions
1. In a theme that supports `layout` (like `tt1-blocks`) insert a `Heading` in a post
2. Insert a `Group` and change the `layout`'s widths (`content` and `wide`)
3. Insert another `Heading` inside this `Group`
4. Check the `alignment control options` in both `Heading` blocks.

In general I will need help with testing to find out missing functionality/info, especially from folks experienced in themes.

## Tasks / Notes
- [ ] Decide on the text to show when `wide alignments` are not supported.
- [ ] Check how to calculate properly if `wide alignments` are supported
- [ ] Can we get `wide and content` widths from somewhere else besides `layout`?
- [x] Check if the align control when is a toolbar (`ToolbarGroup`) is affected somehow or needs changes
- [x] Update the `center` icon with the new one
- [ ] Check mobile
- [ ] Check whether to leave `useAvailableAlignments` intact and incorporate a new helper hook

## Notes
1. I updated the `align` tests also made sure that work as expected in `tt1 and tt1-blocks` as the button context changes. I didn't add to run on both themes though.. Should we run these for both themes? 🤔
